model/llm/llm_test.go

```diff
@@ -340,3 +340,227 @@ func TestLLMCodeRepairSourceFilePrompt(t *testing.T) {
 		`),
 	})
 }
+
+func TestLLMTranspileSourceFilePrompt(t *testing.T) {
+	type testCase struct {
+		Name string
+
+		Data *llmTranspileSourceFilePromptContext
+
+		ExpectedMessage string
+	}
+
+	validate := func(t *testing.T, tc *testCase) {
+		t.Run(tc.Name, func(t *testing.T) {
+			actualMessage, actualErr := llmTranspileSourceFilePrompt(tc.Data)
+			require.NoError(t, actualErr)
+
+			assert.Equal(t, tc.ExpectedMessage, actualMessage)
+		})
+	}
+
+	validate(t, &testCase{
+		Name: "Transpile Go into Java",
+
+		Data: &llmTranspileSourceFilePromptContext{
+			llmSourceFilePromptContext: llmSourceFilePromptContext{
+				Language: &java.Language{},
+
+				Code: bytesutil.StringTrimIndentations(`
+					package com.eval;
+
+					class Foobar {
+						static int foobar(int i) {}
+					}
+				`),
+				FilePath:   "Foobar.java",
+				ImportPath: "com.eval",
+			},
+			OriginLanguage: &golang.Language{},
+			OriginFileContent: bytesutil.StringTrimIndentations(`
+				package foobar
+
+				func foobar(i int) int {
+					return i + 1
+				}
+			`),
+		},
+
+		ExpectedMessage: bytesutil.StringTrimIndentations(`
+			Given the following Go code file, transpile it into a Java code file.
+			The response must contain only the transpiled Go source code and nothing else.
```

Comment:
The response must contain only the transpiled **Java** source code and nothing else.